### ISSUE TITLE
How about move pytorch_transformer to sent_encoder level?

### ISSUE THREAD

**Issue by HaokunLiu** [nyu-mll/jiant#878]
In practice, when ever using pretrained transformer-based models, we always use sent_encoder as "none", null phrase layer, and no highway layer. It's identical to using transformer as sent_encoder directly.

Plus, there have been researches showing BERT, unlike ELMo, does not benefit from more task-specific layers on top, and another paper showing this kind of model is more similar to a fusion of tradition pipeline, rather than simply a word-level embedding module. Therefore, in the near future, there probably won't be any need to pile layers on top of pretrained transformer models.

In addition, this simplify things a lot. If we consider pytorch_transformer module at the same level as sentence encoder, we can directly output the mask in our pytorch_transformer interface. And we no longer need to worry about the strange config terms unrelated to BERT, XLNet etc in sentence_encoder module, when writing config file for tranformer-based experiments.

**Comment by HaokunLiu** [comment]
Related #875 #876

**Comment by sleepinyourhat** [comment]
I disagree—the current logic is confusing, but I think this choice was right. All of the pytorch_transformers models have their own tokenization and input handling, so it's impossible to stack them above other models. However it is possible (and occasionally reasonable) to stack other non-trivial models on top of them.

The longer-term solution to make this clearer should be to simply eliminate the distinction, and allow people to define stacking in the config file however they like.

**Comment by HaokunLiu** [comment]
Ah, I see. So, eventually, we want some mechanism to achieve flexibility while maintaining the user-friendly config-based interface, yes? That's something to think about...
